temporal-sdk/src/main/java/io/temporal/common/interceptors/WorkflowOutboundCallsInterceptor.java

```diff
@@ -50,12 +51,180 @@
  */
 public interface WorkflowOutboundCallsInterceptor {
 
-  final class WorkflowResult<R> {
+  final class ActivityInput<R> {
+    private final String activityName;
+    private final Class<R> resultClass;
+    private final Type resultType;
+    private final Object[] args;
```

Comment:
Rather than using `Object` for these args/payloads it might be nice to introduce some kind of `Serializable` interface the DataConverter could use, pushing some type safety a little higher up.

That said I looked a bit and that'd probably be a huge change so feel free to ignore.

Reply:
Such interface already exists. See [Values](https://github.com/temporalio/sdk-java/blob/master/temporal-sdk/src/main/java/io/temporal/common/converter/Values.java). The reason we pass objects through the interceptor is to give it chance to look at them. As the interceptor doesn't know the types of arguments, passing through arguments as Values is useless.